virtual_ecosystem/models/animal/animal_model.py

```diff
@@ -489,16 +491,22 @@ def calculate_litter_additions_from_herbivory(self) -> dict[str, DataArray]:
 
         # Find the size of the leaf waste pool (in carbon terms)
         leaf_addition = [
-            self.leaf_waste_pools[cell_id].mass_current / self.data.grid.cell_area
+            self.leaf_waste_pools[cell_id].mass_cnp["carbon"] / self.data.grid.cell_area
             for cell_id in self.data.grid.cell_id
         ]
         # Find the chemistry of the pools as well
         leaf_c_n = [
-            self.leaf_waste_pools[cell_id].c_n_ratio
+            self.leaf_waste_pools[cell_id].mass_cnp["carbon"]
+            / self.leaf_waste_pools[cell_id].mass_cnp["nitrogen"]
+            if self.leaf_waste_pools[cell_id].mass_cnp["nitrogen"] > 0
+            else 0.0
             for cell_id in self.data.grid.cell_id
         ]
         leaf_c_p = [
-            self.leaf_waste_pools[cell_id].c_p_ratio
+            self.leaf_waste_pools[cell_id].mass_cnp["carbon"]
+            / self.leaf_waste_pools[cell_id].mass_cnp["phosphorus"]
+            if self.leaf_waste_pools[cell_id].mass_cnp["phosphorus"] > 0
+            else 0.0
```

Comment:
Sorry despite what I said earlier using 0.0 here doesn't really work, but obviously using `inf` for the divide by zero case would break the downstream calculations. Not 100% sure what the best approach is here, I guess this is a value that you should never expect, so maybe we just accept this as a weird edge case?

Reply:
Okay actually think you want to add the following between the if and the else
```
elif self.leaf_waste_pools[cell_id].mass_cnp["carbon"] > 0 np.inf```